// @padcom/vue-i18n/dist/index.mjs
import { ref as i, provide as g, inject as m, getCurrentInstance as x, watch as E } from "vue";
function f(l = "en") {
  var e;
  return ((e = globalThis.navigator) == null ? void 0 : e.language.split("-")[0]) || l;
}
function N(l, e) {
  return Object.getOwnPropertyNames(e).map((r) => typeof e[r] == "string" ? { [[...l, r].join(".")]: e[r] } : N([...l, r], e[r])).reduce((r, o) => ({ ...r, ...o }), {});
}
function v(l) {
  const e = {};
  return Object.getOwnPropertyNames(l).forEach((r) => {
    const o = N([], l[r]);
    Object.getOwnPropertyNames(o).forEach((a) => {
      e[a] = e[a] || {}, e[a][r] = o[a];
    });
  }), e;
}
function $(...l) {
  const e = {};
  return l.forEach((r) => {
    Object.getOwnPropertyNames(r).forEach((o) => {
      e[o] = e[o] || {}, Object.getOwnPropertyNames(r[o]).forEach((a) => {
        e[o][a] = r[o][a];
      });
    });
  }), e;
}
function j(l, e, r) {
  return l === "global" ? r : l === "local-first" ? $(r, e) : e;
}
function d(l, e, r, o) {
  return o[l] ? o[l][e] ? { locale: e, message: o[l][e] } : o[l][r] ? { locale: r, message: o[l][r] } : { locale: f(), message: l } : { locale: f(), message: l };
}
const h = /{\s*(\w+)\s*}/g;
function w(l) {
  return (e, r) => l.hasOwnProperty(r) ? l[r] : e;
}
function T(l, e) {
  if (e === 0)
    return 0;
  if (e === 1)
    return 0;
  if (e === 2)
    switch (l) {
      case 1:
        return 0;
      default:
        return 1;
    }
  else
    switch (l) {
      case 0:
        return 0;
      case 1:
        return 1;
      default:
        return 2;
    }
}
function I(l, e, r) {
  const o = r[l.locale] || T, a = l.message.split("|").map((t) => t.trim());
  return a[o(e, a.length)];
}
const y = Symbol("vue-i18n-pluralization-rules"), S = Symbol("vue-i18n-translations"), z = Symbol("vue-i18n-fallback-locale"), R = Symbol("vue-i18n-locale"), c = {};
function D({
  locale: l = f(),
  fallbackLocale: e = f(),
  messages: r = {},
  pluralizationRules: o = {}
} = {}) {
  if (c.locale || c.fallbackLocale || c.translations)
    throw new Error("vue-i18n already initialized");
  return c.locale = i(l), c.fallbackLocale = i(e), c.translations = j("global", v({}), v(r)), c.pluralizationRules = o || {}, {
    get locale() {
      return c.locale.value;
    },
    set locale(a) {
      c.locale.value = a;
    },
    get fallbackLocale() {
      return c.fallbackLocale.value;
    },
    set fallbackLocale(a) {
      c.fallbackLocale.value = a;
    },
    get availableLocales() {
      return Object.getOwnPropertyNames(r);
    },
    t(a, t = {}) {
      const { message: p } = d(a, this.locale, this.fallbackLocale, c.translations);
      return p.replace(h, w(t));
    },
    tc(a, t, p = {}) {
      const s = d(a, this.locale, this.fallbackLocale, c.translations);
      return I(s, t, o).replace(h, w(p));
    }
  };
}
function F({
  locale: l = f(),
  fallbackLocale: e = f(),
  messages: r = {},
  pluralizationRules: o = {}
}) {
  const a = i(l);
  g(R, i(a));
  const t = i(e);
  return g(z, i(t)), g(S, v(r)), g(y, o), {
    locale: a,
    fallbackLocale: t
  };
}
function H({
  useScope: l = "local-first"
} = {}) {
  var L, O;
  const e = c.locale || m(R), r = c.fallbackLocale || m(z), o = c.pluralizationRules || m(y), a = c.translations || m(S), t = x(), p = (
    // Vue 3
    ((L = t == null ? void 0 : t.type) == null ? void 0 : L.i18n) || // Vue 2
    ((O = t == null ? void 0 : t.proxy) == null ? void 0 : O.$options.i18n) || // Fallback
    {}
  );
  if (!(e && r && a && o))
    throw new Error("vue-i18n not initialized. Either call createI18n() or createI18nContext()");
  const s = j(l, v(p), a);
  return E(e, () => {
    var n, u;
    t != null && t.update && (t == null || t.update()), (n = t == null ? void 0 : t.proxy) != null && n.$forceUpdate && ((u = t == null ? void 0 : t.proxy) == null || u.$forceUpdate());
  }), E(r, () => {
    var n, u;
    t != null && t.update && (t == null || t.update()), (n = t == null ? void 0 : t.proxy) != null && n.$forceUpdate && ((u = t == null ? void 0 : t.proxy) == null || u.$forceUpdate());
  }), {
    fallbackLocale: r,
    locale: e,
    get availableLocales() {
      const n = /* @__PURE__ */ new Set();
      return Object.getOwnPropertyNames(s).forEach((u) => {
        Object.getOwnPropertyNames(s[u]).forEach((b) => {
          n.add(b);
        });
      }), [...n];
    },
    t(n, u = {}) {
      const { message: b } = d(n, e.value, r.value, s);
      return b.replace(h, w(u));
    },
    tc(n, u, b = {}) {
      const P = d(n, e.value, r.value, s);
      return I(P, u, o).replace(h, w(b));
    }
  };
}
export {
  z as VueI18NFallbackLocaleSymbol,
  R as VueI18NLocaleSymbol,
  y as VueI18NPluralizationRulesSymbol,
  S as VueI18NTranslationsSymbol,
  F as createI18Context,
  D as createI18n,
  f as getAgentLocale,
  H as useI18n
};
